examples/plugins/date/main.go

```diff
@@ -53,11 +54,14 @@ func DateCommand() (*cobra.Command, error) {
 		},
 	}
 
-	// default validation provided by validator package
-	validationErr := validator.Validate(cmd)
-	if validationErr != nil {
-		return nil, validationErr
+	var r validator.Rule = &validator.LengthRule{
```

Comment:
This approach as number of drawbacks:

- it is fixed to the point where we need to redeclare every field 
- cannot be declarative
- Doesn't provide us base for adding other rules (missing interface for rule)